internal/controller/nonadminbackup_controller.go

```diff
@@ -175,19 +173,10 @@ func (r *NonAdminBackupReconciler) setStatusAndConditionForDeletionAndCallDelete
 			return false, err
 		}
 		logger.V(1).Info("NonAdminBackup status marked for deletion")
-		requeueRequired = true // Requeue to ensure latest NAB object in the next reconciliation steps
 	} else {
 		logger.V(1).Info("NonAdminBackup status unchanged during deletion")
 	}
-	if nab.ObjectMeta.DeletionTimestamp.IsZero() {
-		logger.V(1).Info("Marking NonAdminBackup for deletion", constant.NameString, nab.Name)
-		if err := r.Delete(ctx, nab); err != nil {
```

Comment:
once this is called, update predicate is triggered. So, this should be last step to avoid running multiple reconciles

Reply:
But if you just return `false, nil` then reconcile will enter another step and will not exit as last step?

Reply:
I think we may need a 3rd state as it was in the very initial implementation which means "exit cleanly now" to avoid complexity moving parts to the steps where they don't belong. Moving this to the last step e.g. `removeNabFinalizerUponVeleroBackupDeletion` also doesn't makes sense to me as this step is actually to remove finalizer and not call delete on an object.

Reply:
what I did was add this as the last step of deletion, so `return false` will not do anything more

Reply:
How this works with the path where Deletion happens via `DeleteBackupRequest`

Reply:
it should be as it was, only change was the order of steps